port-mirroring: new package for copying network packets

mmaraya opened the pull request:
This is a new package that allows the user to copy network packets from one switch port to another without having to use iptables TEE. This is my first pull request for OpenWrt packages, so please let me know if I missed anything. Thanks!

Signed-off-by: Mike Maraya <mike.maraya@gmail.com>

thess commented:
Thanks for this - A couple of items should be addressed:
* Makefile needs OpenWrt (or your) copyright. Example:

> \#
> \# Copyright (C) 2015 OpenWrt.org
> \#
> \# This is free software, licensed under the GNU General Public License v2.
> \# See /LICENSE for more information.
> \#

* You should not need the postinst/prerm functions for your init script.

mmaraya commented:
Changes done, thanks for all the help!

Signed-off-by: Mike Maraya mike.maraya@gmail.com

thess commented:
Not quite what I had in mind for a copyright on the OpenWrt package (not the library/application) as defined by the OpenWrt Makefile. The copyright should reflect the correct date (it's not 2012 anymore) and there is no need to repeat the license text in the Makefile. Look at other packages for good examples.

Also, you did not sign-off your new commit or give it a proper commit message prefixed with the package name. I suggest when you have corrected things, you should rebase/squash the PR to a single commit (use `git rebase -i ...`).  Then update the PR with `git push -f` to overwrite the original commit in your mirror and the PR will be updated automatically.

mmaraya commented:
Thanks for the quick response! I removed the license text, prefixed/signed-off on the commit message, and squashed the commits together. However, I am not the copyright holder and I am not comfortable changing the copyright notice. This seems to be the accepted approach according to http://stackoverflow.com/questions/2390230/do-copyright-dates-need-to-be-updated.

thess commented:
OpenWrt copyright - I think you are confusing the actual software application with the OpenWrt Makefile and package. You do not necessarily need to be the copyright holder of the Makefile even though it appears you are the author and maintainer. We would prefer to have the package Makefile contain an up-to-date copyright and belonging to OpenWrt (community property). However, the name you currently have on the Makefile is neither the author nor the maintainer/proprietor.

mmaraya commented:
Bruce Geng @gengw2000 authored both the actual software application and the OpenWrt Makefile in 2012 and published on https://code.google.com/p/port-mirroring/. Bruce no longer maintains the software so I exported the code to https://github.com/mmaraya/port-mirroring in September 2015 and have been maintaining it since. I've sent Bruce an email asking if he's okay with transferring the copyright to OpenWrt. He's been really cool about other changes I've proposed and so I expect he'll be okay with transferring copyright to OpenWrt. I'll update the Makefile and squash the commits once I hear back.

Other than the copyright notice, does everything else look okay?

thess commented:
Yup - looks OK

mmaraya commented:
@gengw2000 wrote back that he was fine with OpenWrt holding the copyright for the Makefile.  e428fd92f0066430a255b115107720818c7e34b8 ready for review.

thess commented:
Thanks
